src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/Services/SemanticTokensEditsDiffer.cs

```diff
@@ -67,37 +67,59 @@ private Container<SemanticTokensEdit> ProcessEdits(IReadOnlyList<DiffEdit> diffs
                         }
                         else
                         {
-                            results.Add(new SemanticTokensEdit
+                            razorResults.Add(new RazorSemanticTokensEdit
                             {
                                 Start = diff.Position,
-                                Data = ImmutableArray<int>.Empty,
                                 DeleteCount = 1,
                             });
                         }
                         break;
                     case DiffEdit.Type.Insert:
                         if (current != null &&
-                            current.Data.HasValue &&
-                            current.Data.Value.Any() &&
+                            current.Data != null &&
+                            current.Data.Any() &&
```

Comment:
Can avoid allocating an enumerator:

```suggestion
                            current.Data.Count > 0 &&
```

Reply:
Would the compiler handle this optimization possibly?

Reply:
Doubt it but possible. They have other special cases in other locations. Could always build a console app in debug/release and compare the IL

Reply:
No compiler optimization it seems:

![image](https://user-images.githubusercontent.com/14852843/109231101-cb922480-777a-11eb-988c-291ee476fd6b.png)

![image](https://user-images.githubusercontent.com/14852843/109231163-e6fd2f80-777a-11eb-93fe-2f08a9eeecdb.png)